modules/infra/control/loopback.c

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr.h>
+#include <gr_control_input.h>
+#include <gr_control_output.h>
+#include <gr_eth.h>
+#include <gr_iface.h>
+#include <gr_infra.h>
+#include <gr_log.h>
+#include <gr_loopback.h>
+#include <gr_mempool.h>
+#include <gr_module.h>
+
+#include <event2/event.h>
+#include <rte_errno.h>
+#include <rte_malloc.h>
+
+#include <fcntl.h>
+#include <linux/if_tun.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define TUN_TAP_DEV_PATH "/dev/net/tun"
+
+static struct rte_mempool *loopback_pool;
+static struct event_base *global_ev_base;
+
+struct iface_info_loop {
+	int fd;
+	struct event *ev;
+};
+
+void loopback_tx(struct rte_mbuf *m) {
+	struct mbuf_data *d = mbuf_data(m);
+	struct iface_info_loop *lo;
+	void *data;
+
+	lo = (struct iface_info_loop *)d->iface->info;
+	if (rte_pktmbuf_linearize(m) == 0) {
+		data = rte_pktmbuf_mtod(m, char *);
```

Comment:
Nit: data is void *, so call mtod accordingly.

Reply:
Modified data to char* as need to access to fill in struct tun_pi.